src/AP1roG/AP1roG.cpp

```diff
@@ -24,4 +24,35 @@ AP1roG::AP1roG(const GQCG::AP1roGGeminalCoefficients& geminal_coefficients, doub
 {}
 
 
+
+/*
+ *  HELPER FUNCTIONS
+ */
+/**
+ *  Calculate the AP1roG energy given AP1roG geminal coefficients @param G and Hamiltonian parameters @param ham_par
+ */
+double calculateAP1roGEnergy(const GQCG::AP1roGGeminalCoefficients& G, const GQCG::HamiltonianParameters& ham_par) {
+
+    Eigen::MatrixXd h_SO = ham_par.get_h().get_matrix_representation();
+    Eigen::Tensor<double, 4> g_SO = ham_par.get_g().get_matrix_representation();
+
+
+    // KISS implementation of the AP1roG energy
+    double E = 0.0;
+    for (size_t j = 0; j < G.get_N_P(); j++) {
+        E += 2 * h_SO(j,j);
```

Comment:
Does h_SO(j,j) require to be h_SO(j, j) ? if so, repeat through entire code.

Reply:
I don't think we have to add whitespace to element-accessors for matrices like here.